utility/src/main/java/org/oppia/android/util/parser/html/HtmlParser.kt

```diff
@@ -66,6 +66,10 @@ class HtmlParser private constructor(
       htmlContentTextView.movementMethod = LinkMovementMethod.getInstance()
     }
 
+    if (isRtlLayout(htmlContentTextView)) {
+      htmlContentTextView.textDirection = View.TEXT_DIRECTION_ANY_RTL
+    }
```

Comment:
Please add a single line comment contextualizing why it's correct to set the text direction in this case.

Also, do we need to change it back if RTL is turned off mid-layout?

Finally, can you add a test to verify that this is correct set/not set in the expected situations?

Reply:
Added comments and testcases.

Reply:
> Also, do we need to change it back if RTL is turned off mid-layout?

No, it does the changes automatically when the layout is changed to LTR. Added test for this case too.